src/components/input/input.tsx

```diff
@@ -511,6 +507,8 @@ export class Input implements LabelableComponent, FormComponent, InteractiveComp
       return;
     }
     if (event.key === "ArrowUp") {
+      /* prevent default behavior of moving cursor to the beginning of the input when holding down ArrowUp */
+      event.preventDefault();
```

Comment:
Having these two separate keyDownHandlers threw me off recently too.  Will be nice when we get around to separating out the number component.